### PR TITLE
Update README and add arch diagram

### PR DESCRIPTION
**Pull request by yangl900**
Update README with architecture diagram and explain how it works.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
 
+# MSI ACR Pull
+MSI ACR Pull enables deployments in a Kubernetes cluster to use any user assigned managed identity to pull images from Azure container registry. With this, each application can use its own identity to pull container images.
+
+# How it works
+The architecture looks like below. As an user you will create a custom resource `ACRPullBinding`, which binds a managed identity (using client ID or resource ID) to an Azure container registry (using its FQDN). 
+
+Internally, the `ACRPullBindingController` watches the `ACRPullBinding` resource, and for each of them, create a secret in the namespace. The secret content is a Docker image pull config, and the password is the ACR access token that the controller exchanged from ACR using managed identity. The secret will be refreshed 30min before it expire automatically. The controller will also associate the secret to the specified service account in namespace (by default, use the default service account). With this, any pods created in the namespace will automatically pull images from the ACR using the specified managed identity credential.
```
Comment (weng271190436) [inline README.md:8]: I don't think we have done the part where the controller associates the secret to the specified service account or the default service account yet but it should be there by the end of this week

File: README.md [hunk 2]
```diff
@@ -1,4 +1,14 @@
 
+# MSI ACR Pull
+MSI ACR Pull enables deployments in a Kubernetes cluster to use any user assigned managed identity to pull images from Azure container registry. With this, each application can use its own identity to pull container images.
+
+# How it works
+The architecture looks like below. As an user you will create a custom resource `ACRPullBinding`, which binds a managed identity (using client ID or resource ID) to an Azure container registry (using its FQDN).
```
Comment (weng271190436) [inline README.md:6]: the resource id part is also not there yet I think? Not blocking because it will be there pretty soon

File: README.md (hunk 1 above)
Comment (yangl900) [reply README.md:8]: yep. trying to catch the goal state of initial release.

File: README.md (hunk 2 above)
Comment (yangl900) [reply README.md:6]: yep, trying to describe the goal state.

**Pull request merged**